### DISCUSSION TITLE
MultiTask empty label problem

### DISCUSSION THREAD

**Discussion by peng-peng0913** [open-mmlab/mmclassification#1395] [Q&A] [answered]
I have been doing MultiTask recently, but I have encountered a problem.
My data.list
----------------------------------------------------------------
image name | label (task_1) | label (task_2) | label (task_3) |
----------------------------------------------------------------
xxx1.jpg               0                     2                        1
xxx2.jpg               -1                   1                         0
----------------------------------------------------------------
If the label is - 1, it means that the image is not labeled. Direct training will report an error. How can I handle the - 1 label to match the input.

**Comment by mzr1996** [comment, marked as answer]
Hello, here are two methods:
1. Format your annotation file in the `MultiTaskDataset` required format.
https://github.com/open-mmlab/mmclassification/blob/654a337d3e93364328044d110b25441d907f4bd6/mmcls/datasets/multi_task.py#L32-L58
2. Implement your own dataset class to handle your annotation file, you can find tutorial [here](https://mmclassification.readthedocs.io/en/dev-1.x/advanced_guides/datasets.html), and the returned info should be like
```python
[{'img_path': xxx1.jpg,
  'gt_label': {
      'task1': 0,
      'task2': 2,
      'task3': 1,
     }
 },
 {'img_path': xxx2.jpg,
  'gt_label': {
      'task2': 1,
      'task3': 0,
     }
 }
 ...
]
```

**Reply by peng-peng0913** [reply]
I've been using mmcls version 0.24 and writing my own MultiTask. I now see that the new version of mmcls supports MultiTask. I will update my mmcls version. Thank you.
